client/web/src/insights/pages/search-insight-creation/components/form-series-input/FormSeriesInput.tsx

```diff
@@ -0,0 +1,192 @@
+    onCancel?: () => void
+}
+
+/** Displays form series input (three field - name field, query field and color picker). */
+export const FormSeriesInput: React.FunctionComponent<FormSeriesProps> = props => {
+    const {
+        name,
+        query,
+        color,
+        className,
+        cancel = false,
+        autofocus = false,
+        onCancel = noop,
+        onSubmit = noop,
+        innerRef,
+    } = props
+
+    const hasNameControlledValue = !!name
+    const hasQueryControlledValue = !!query
+
+    const { handleSubmit, form } = useForm<FormSeriesValues>({
+        initialValues: {
+            seriesName: name,
+            seriesQuery: query,
+            seriesColor: color ?? DEFAULT_ACTIVE_COLOR,
+        },
+        onSubmit: values =>
+            onSubmit({
+                name: values.seriesName,
+                query: values.seriesQuery,
+                color: values.seriesColor,
+            }),
+    })
+
+    const nameField = useField('seriesName', form, requiredNameField)
+    const queryField = useField('seriesQuery', form, validQuery)
+    const colorField = useField('seriesColor', form)
+
+    const nameReference = useRef<HTMLInputElement>(null)
+    const queryReference = useRef<HTMLInputElement>(null)
+
+    // In case if consumer asked this component to be focused (call .focus() on ref)
+    // We focus first invalid field. Otherwise we focus first field of form series
+    // - series name field.
+    useImperativeHandle(innerRef, () => ({
+        focus: () => {
+            if (nameField.meta.error) {
+                return nameReference.current?.focus()
+            }
+
+            if (queryField.meta.error) {
+                return queryReference.current?.focus()
+            }
+
+            nameReference.current?.focus()
+        },
+    }))
+
+    useEffect(() => {
+        if (autofocus) {
+            nameReference.current?.focus()
+        }
+    }, [autofocus])
+
+    const handleSubmitButton = useCallback(
+        async (event: React.MouseEvent) => {
+            if (nameField.meta.error) {
+                event.preventDefault()
+                return nameReference.current?.focus()
+            }
+
+            if (queryField.meta.error) {
+                event.preventDefault()
+                return queryReference.current?.focus()
+            }
+
+            // handleSubmit work with form element and use form event
+            // but we can't have sub forms for the sake of semantics.
+            // if this case synthetic event of totally comparable with event
+            // from submit button.
+            await handleSubmit((event as unknown) as React.SyntheticEvent<HTMLFormElement>)
+        },
+        [handleSubmit, nameField.meta.error, queryField.meta.error]
+    )
+
+    return (
+        <div className={classnames(styles.formSeriesInput, className)}>
+            <InputField
+                title="Name"
+                placeholder="ex. Function component"
```

Comment:
According to our [content guidelines](https://about.sourcegraph.com/handbook/communication/content_guidelines/style_and_mechanics#abbreviations-acronyms-latinisms-jargon), we should use complete English words and avoid abbreviations
```suggestion
                placeholder="Example: Function components"
```